src/develop/masks/brush.c

```diff
@@ -1220,8 +1226,10 @@ static int dt_brush_events_button_pressed(struct dt_iop_module_t *module, float
       if(!gpt) return 0;
       // we start the form dragging
       gui->source_dragging = TRUE;
+      dt_pthread_mutex_lock(&darktable.develop->preview_pipe_mutex);
```

Comment:
for the backbuf_width/height ones, preview_pipe->backbuf_mutex would be enough to lock (see pixelpipe_hb.c:2440)